clang/utils/TableGen/ClangProgModelBuiltinEmitter.cpp

```diff
@@ -237,6 +245,66 @@ class BuiltinNameEmitter {
   // same entry (<I1, I2, I3>).
   MapVector<BuiltinIndexListTy *, BuiltinTableEntries> SignatureListMap;
 };
+
+/// Generate a json file representing the builtins as understood by clang.
+/// This allows the generation of a library interface.
+class JSONBuiltinInterfaceEmitter : public BuiltinNameEmitter {
+  struct TypeDesc {
+    TypeDesc(const Record *T);
+    // Return a string representing the "base" type (ignores pointers).
+    std::string GetBaseTypeAsStr() const;
+
+    enum UnderlyingType { BOOL, INT, UINT, FLOAT, BUILTIN };
```

Comment:
Typically we want to use either a scoped enum, or make those names unique (UT_BOOL).

Reply:
made it scoped